### PR TITLE
Fix ShardEvents and ShardBulkDocs null metrics

### PR DESCRIPTION
**Pull request by Tjofil**
Signed-off-by: Filip Drobnjakovic <drobnjakovicfilip@gmail.com>

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Issue #251 

**Describe the solution you are proposing**
Did a revert to a previous known functioning point. Both ShardEvents and ShardBulkDocs should now display correct, non-null values.
Currently tested only by hand comparing responses of the patched and non-patched docker OS cluster.

**Describe alternatives you've considered**

**Additional context**

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kkhatua** [comment]
@Tjofil  is there a test to verify this? Or did you repro in the original docker image and verified it as fixed in the patched docker image? 
Would be good to know why it broke and perhaps add a test around this. (More tests for other metrics can be added in the future).

**Comment by Tjofil** [comment]
Hey @kkhatua 
Changed the PR comment to describe the current level of testing. I agreed with @khushbr to fix it now and have it ready for the next release and maybe do further investigation later, so I just made a init PR and am waiting for her comment on the next steps and potential testing. If you have some suggestions I'd be happy to add them.

**Comment by khushbr** [comment]
> Hey @kkhatua Changed the PR comment to describe the current level of testing. I agreed with @khushbr to fix it now and have it ready for the next release and maybe do further investigation later, so I just made a init PR and am waiting for her comment on the next steps and potential testing. If you have some suggestions I'd be happy to add them.

Thank you @Tjofil for the quick fix. Let us add a unit test  for `shardBulk` under [ShardStateMetricsSnapshotTest.java](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/src/test/java/org/opensearch/performanceanalyzer/reader/ShardStateMetricsSnapshotTest.java) with this PR. 

Next, we can further beef up our unit test as part of Issue https://github.com/opensearch-project/performance-analyzer-rca/issues/284

File: src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java [hunk 1]
```diff
@@ -53,6 +54,42 @@ public void before() throws Exception {
...
     }
 
+    @Test
+    public void testShardBulkParseAndQuery() throws Exception {
+        deleteAll();
+        ReaderMetricsProcessor mp =
+                new ReaderMetricsProcessor(rootLocation, true, new AppContext());
+
+        mp.processMetrics(rootLocation, 1566413985000L);
+        mp.processMetrics(rootLocation, 1566413990000L);
```
Comment (khushbr) [inline src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:64]: I don;t see any `shardbulk` metric within the timestamp file below. How is this test working? 

```
^threads/21543/shardbulk/23632396/start
current_time:1674690064926
StartTime:1674690064926
ItemCount:8
IndexName:index-1
ShardID:2
```

File: src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java [hunk 2]
```diff
@@ -53,6 +54,42 @@ public void before() throws Exception {
...
+                                Arrays.asList("Operation"));
+
+        boolean shardbulkEncountered = false;
+
+        for (Record record : res) {
+            if (PerformanceAnalyzerMetrics.sShardBulkPath.equals(record.get("Operation"))) {
+                assertNotNull(record.get("ShardEvents"));
+                assertNotNull(record.get("ShardBulkDocs"));
+                assertTrue((Double) record.get("ShardEvents") >= 1.0);
+                assertTrue((Double) record.get("ShardBulkDocs") >= 1.0);
```
Comment (khushbr) [inline src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:81]: Asserting for `>= 1.0` checks for missing data but not for the accuracy. Let us assert for value of these `ShardEvents` and `ShardBulkDocs`

Can use [AdmissionControlSnapshotTest](https://github.com/opensearch-project/performance-analyzer-rca/blob/4734d6d5ba68ba41b6c62ab2cbf5ff22aa52a186/src/test/java/org/opensearch/performanceanalyzer/reader/AdmissionControlSnapshotTest.java) as reference

File: src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java (hunk 1 above)
Comment (Tjofil) [reply src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:64]: The `shardbulk` metrics are in the first one (which was already in resources folder). The second processMetrics invocation is for the ReaderMetricsProcessor to emit the Metrics from the previous run. On each run, emitMetrics emits Snapshots from currWindowStartTime - MetricsConfiguration.SAMPLING_INTERVAL timestamp.
https://github.com/opensearch-project/performance-analyzer-rca/blob/main/src/main/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessor.java#L361

File: src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java (hunk 2 above)
Comment (Tjofil) [reply src/test/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java]: Agreed, I've put the exact expected values.

**Review by khushbr — APPROVED** [review]

**Review by kkhatua — APPROVED** [review]

**Comment by systemosyn** [comment]
Hey @Tjofil and @khushbr,
Do you think this issue was the root cause of this one: https://github.com/opensearch-project/performance-analyzer/issues/77#issuecomment-1310019450 
Where I saw that only the shabulk started event are written in the EventLog Files. If yes, maybe we could close this issue as well, right?

**Pull request merged**